### PR TITLE
[PyTorch] Merge `k_channels` and `v_channels` back to `kv_channels`

### PR DESCRIPTION
**Pull request by cyanguwa**
# Description

This PR merges `k_channels` and `v_channels` back to `kv_channels` in `DotProductAttention.__init__()` to avoid breaking backward compatibility, but it allows users to pass in a tuple `(int, int)` in MLA (multi-latent attention) cases.

We will add support for MLA in `MultiHeadAttention` and `TransformerLayer` in future PRs.

## Type of change

- [ ] Documentation change (change only to the documentation, either a fix or a new content)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Infra/Build change
- [ ] Code refractor

## Changes

Please list the changes introduced in this PR:

- Merge `k_channels` and `v_channels` back to `kv_channels` in `DotProductAttention`

# Checklist:

- [x] I have read and followed the [contributing guidelines](https://github.com/NVIDIA/TransformerEngine/blob/main/CONTRIBUTING.rst)
- [x] The functionality is complete
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: transformer_engine/pytorch/attention.py [hunk 1]
```diff
@@ -5310,8 +5308,12 @@ def __init__(
         self.cp_global_ranks = cp_global_ranks
         self.cp_stream = cp_stream
 
-        self.hidden_size_per_attention_head = k_channels
-        self.v_channels = k_channels if v_channels is None else v_channels
+        self.hidden_size_per_attention_head_k = (
```
Comment (ptrendx) [inline transformer_engine/pytorch/attention.py:5311]: Is this unused? How could `self.hidden_size_per_attention_head` and `self.v_channels` change to `self.hidden_size_per_attention_head_k` and `self.hidden_size_per_attention_head_v` without any other changes in the code?

Comment (cyanguwa) [reply transformer_engine/pytorch/attention.py:5311]: It wasn't used, until I added that shape check later (see [here](https://github.com/cyanguwa/TransformerEngine/blob/ed8ed8ccbb941f94906106ee372566337d2d69d0/transformer_engine/pytorch/attention.py#L5635)). I can delete all this together but I think it's good to have that shape check as well.

**Review by ptrendx — APPROVED** [review]

**Pull request merged**